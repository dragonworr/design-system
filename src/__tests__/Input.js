import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { Input } from '..'

describe('Input', () => {
  test('it renders', () => {
    const json = renderer.create(<Input />).toJSON()
    expect(json).toMatchSnapshot()
  })
  test('it renders an input element with a red border with a color prop is passed', () => {
    const json = renderer.create(<Input color="red" />).toJSON()
    expect(json).toMatchSnapshot()
  })
  test('it renders an input element with a really large padding and margin', () => {
    const json = renderer.create(<Input p={4} m={4} />).toJSON()
    expect(json).toMatchSnapshot()
  })
  test('it renders an input element with large text', () => {
    const json = renderer.create(<Input fontSize={4} />).toJSON()
    expect(json).toMatchSnapshot()
  })
})
