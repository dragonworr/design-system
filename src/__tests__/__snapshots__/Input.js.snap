// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input it renders 1`] = `
.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  font-size: 16px;
  background-color: transparent;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #abb7c2;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  border-color: #abb7c2;
  box-shadow: 0 0 0 0 #abb7c2;
}

.c1::-webkit-input-placeholder {
  color: #abb7c2;
}

.c1::-moz-placeholder {
  color: #abb7c2;
}

.c1:-ms-input-placeholder {
  color: #abb7c2;
}

.c1::placeholder {
  color: #abb7c2;
}

.c1::-ms-clear {
  display: none;
}

.c1:focus {
  outline: 0;
  border-color: #007aff;
  box-shadow: 0 0 0 1px #007aff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c0 * {
  box-sizing: border-box;
}

<div
  className="c0"
>
  <input
    className="c1"
  />
</div>
`;

exports[`Input it renders an input element with a really large padding and margin 1`] = `
.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  font-size: 16px;
  background-color: transparent;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #abb7c2;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  border-color: #abb7c2;
  box-shadow: 0 0 0 0 #abb7c2;
  margin: 32px;
  padding: 32px;
}

.c1::-webkit-input-placeholder {
  color: #abb7c2;
}

.c1::-moz-placeholder {
  color: #abb7c2;
}

.c1:-ms-input-placeholder {
  color: #abb7c2;
}

.c1::placeholder {
  color: #abb7c2;
}

.c1::-ms-clear {
  display: none;
}

.c1:focus {
  outline: 0;
  border-color: #007aff;
  box-shadow: 0 0 0 1px #007aff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c0 * {
  box-sizing: border-box;
}

<div
  className="c0"
>
  <input
    className="c1"
  />
</div>
`;

exports[`Input it renders an input element with a red border with a color prop is passed 1`] = `
.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  font-size: 16px;
  background-color: transparent;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #abb7c2;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  border-color: #c00;
  box-shadow: 0 0 0 0 #c00;
}

.c1::-webkit-input-placeholder {
  color: #abb7c2;
}

.c1::-moz-placeholder {
  color: #abb7c2;
}

.c1:-ms-input-placeholder {
  color: #abb7c2;
}

.c1::placeholder {
  color: #abb7c2;
}

.c1::-ms-clear {
  display: none;
}

.c1:focus {
  outline: 0;
  border-color: #c00;
  box-shadow: 0 0 0 1px #c00;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c0 * {
  box-sizing: border-box;
}

<div
  className="c0"
>
  <input
    className="c1"
    color="red"
  />
</div>
`;

exports[`Input it renders an input element with large text 1`] = `
.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  font-size: 16px;
  background-color: transparent;
  border-radius: 2px;
  border-width: 1px;
  border-style: solid;
  border-color: #abb7c2;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  border-color: #abb7c2;
  box-shadow: 0 0 0 0 #abb7c2;
}

.c1::-webkit-input-placeholder {
  color: #abb7c2;
}

.c1::-moz-placeholder {
  color: #abb7c2;
}

.c1:-ms-input-placeholder {
  color: #abb7c2;
}

.c1::placeholder {
  color: #abb7c2;
}

.c1::-ms-clear {
  display: none;
}

.c1:focus {
  outline: 0;
  border-color: #007aff;
  box-shadow: 0 0 0 1px #007aff;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c0 * {
  box-sizing: border-box;
}

<div
  className="c0"
>
  <input
    className="c1"
    fontSize={4}
  />
</div>
`;
